### ISSUE TITLE
Unexpected result on keras.ops.nn.softmax when running with tuple axis

### ISSUE THREAD

**Issue by shashaka** [keras-team/keras#20021]
I found that tuple axis case is implemented on softmax fucntion. 
However, it shows unexpected result when I run the test case with tuple axis parameter.
I think if softmax function was applied to input along axis, the sum of all result should be 1.0.,but, it was not at the test result.

I don't know exactly about the history related in this implementation, however, I think it should be fixed or can be removed,
because, this axis parameter's type is declared as integer on documentation.

- keras/src/ops/nn.py
```
def softmax(x, axis=-1):
...

if isinstance(axis, tuple):
    original_shape = x.shape
    new_shape = []
    skip_dims = set(axis)
    i = 0
    while i < len(original_shape):
        if i in skip_dims:
            size = 1
            while i in skip_dims:
                size *= original_shape[i]
                i += 1
            new_shape.append(size)
        else:
            new_shape.append(original_shape[i])
            i += 1
    x = backend.numpy.reshape(x, new_shape)
    x = backend.nn.softmax(x, axis=-1)
    x = backend.numpy.reshape(x, original_shape)
    return x
```

- test case [ keras/src/ops/nn_test.py ]
```
def test_softmax_correctness_with_axis_tuple(self):
    input = np.array([[[1.0, 2.0], [3.0, 4.0]], [[5.0, 6.0], [7.0, 8.0]]])
    combination = combinations(range(3), 2)
    for axis in list(combination):
        result = knn.softmax(input, axis=axis)
        normalized_sum_by_axis = np.sum(result, axis=axis)
        self.assertAllClose(normalized_sum_by_axis, 1.0)
```
